app/commands/solution/search_community_solutions.rb

```diff
@@ -17,25 +17,64 @@ def initialize(exercise, page: nil, per: nil, criteria: nil)
     end
 
     def call
-      @solutions = exercise.solutions.published
+      results = client.search(index: 'solutions', body: search_body)
 
-      filter_criteria!
-      sort!
+      solution_ids = results["hits"]["hits"].map { |hit| hit["_source"]["id"] }
+      solutions = Solution.where(id: solution_ids).
```

Comment:
Before this line, you need to deal with the fact that `solution_ids` might be `empty`, because the query below will break if there are no ids.

I'd probably do `solutions = solution_ids.present? ? .. : []` or something.